### PR TITLE
FIX getExtensionInstance can return null, add a case to handle that

### PR DESCRIPTION
**Pull request by robbieaverill**
Parent issue: https://github.com/silverstripe/cwp-recipe-kitchen-sink/issues/18

### CODE REVIEW

**Comment by robbieaverill** [comment]
This issue is caused by a regression in https://github.com/silverstripe/silverstripe-framework/pull/8465 that only surfaces in some cases. At the moment I can only see it in `Symbiote\AdvancedWorkflow\Tests\WorkflowEmbargoExpiryTest::testDuplicateRemoveEmbargoExpiry` on 5.0.x-dev when queuedjobs 4.x-dev is installed too. I'm looking to see if I can patch the regression as well, but this chance is defensive and should be merged too.

**Comment by robbieaverill** [comment]
Ok, patch fixed for that issue

File: src/Core/Extensible.php [hunk 1]
```diff
@@ -135,6 +135,10 @@ protected function defineExtensionMethods()
                 $this->addCallbackMethod($method, function ($inst, $args) use ($method, $extensionClass) {
                     /** @var Extensible $inst */
                     $extension = $inst->getExtensionInstance($extensionClass);
+                    if (!$extension) {
```
Comment (dhensby) [inline src/Core/Extensible.php]: Under what circumstances would there not be an extension instance?

Comment (robbieaverill) [reply src/Core/Extensible.php]: You can reproduce it with the advancedworkflow comment above - the method can return null though, so this is a defensive patch

**Pull request merged**